File: 1121-divide-array-into-increasing-sequences/1121-divide-array-into-increasing-sequences.java
class Solution {
    public boolean canDivideIntoSubsequences(int[] nums, int k) {
        int group = 1, cur = 0, n = nums.length;
        for(int i = 1; i < n; ++i) {
            cur = nums[i - 1] < nums[i] ? 1 : cur + 1;
            group = Math.max(group, cur);
        }
        return group * k <= n;
    }
}

File: 1121-divide-array-into-increasing-sequences/1121-divide-array-into-increasing-sequences.cpp
class Solution {
public:
    bool canDivideIntoSubsequences(vector<int>& nums, int k) {
        int g = 1, cur = 1, n = nums.size();
        for(int i = 1; i < n; ++i) {
            cur = nums[i - 1] < nums[i] ? 1 : cur + 1;
            g = max(g, cur);
        }
        return g * k <= n;
      
    }
};